SDWebImage/SDWebImageGroup.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * This file is part of the SDWebImage package.
+ * (c) Olivier Poitrey <rs@dailymotion.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+#import <Foundation/Foundation.h>
+
+@interface SDWebImageGroup : NSObject
+
+/**
+ * The group to simulate the behavior of `dispatch_group_t`, the difference is
+ * `dispatch_group_notify` is async when call block, `SDWebImageGroup` support
+ * sync/async, default is sync, you should be careful to prevent dead lock.
+ *
+ * @param queue         The queue to which the notify block is submitted when group completes.
+ * @param isAsync       Sync/Async to call notify block, default is sync. Be careful to prevent dead lock.
+ */
+- (nonnull instancetype)initWithNotifyQueue:(nullable dispatch_queue_t)queue
```

Comment:
Could we make this API, more similiar to `disaptch_group` ? Like the [Swift interface](https://developer.apple.com/documentation/dispatch/dispatchgroup) from Apple...

```objective-c
@interface SDWebImageGroup : NSObject

@property (nonatomic, class, readonly) SDWebImageGroup *group;

- (void)enter;
- (void)leave;
- (void)notify:(dispatch_queue_t)queue block:(dispatch_block_t)block;
- (void)wait;
- (void)waitUntil:(dispatch_time_t)timeout;

@end
```